File: Overlays/ingame/src/data/scoreboardConfig.ts
import FrontEndTeam from "./frontEndTeam";

export default class ScoreboardConfig {
    BlueTeam: FrontEndTeam;
    RedTeam: FrontEndTeam;
    GameTime: number;
    SeriesGameCount: number;

    constructor(o: any) {
        this.BlueTeam = new FrontEndTeam(o.BlueTeam);
        this.RedTeam = new FrontEndTeam(o.RedTeam);
        this.GameTime = o.GameTime;
        this.SeriesGameCount = o.SeriesGameCount;
    }
}

File: LeagueBroadcast/Ingame/Data/Frontend/ScoreboardConfig.cs
using LeagueBroadcast.Common;
using LeagueBroadcast.Common.Controllers;

namespace LeagueBroadcast.Ingame.Data.Frontend
{
    public class ScoreboardConfig
    {
        public FrontEndTeam BlueTeam;
        public FrontEndTeam RedTeam;
        public double GameTime;
        public int SeriesGameCount;

        public bool ShouldSerializeGameTime()
        {
            return ConfigController.Component.Ingame.UseCustomScoreboard;
        }

        public bool ShouldSerializeSeriesGameCount()
        {
            return IngameController.CurrentSettings.TeamStats;
        }
    }
}
